File: 7. Reverse Integer/solution.java
public class Solution {
    public int reverse(int x) {
        // Determine the sign of the number
        int numSign = (x < 0) ? -1 : 1;
        
        // Convert the absolute value of the number to a string
        String str = Integer.toString(Math.abs(x));
        
        // Reverse the string
        String reversedStr = new StringBuilder(str).reverse().toString();
        
        // Convert the string back to an integer
        long newNum;
        try {
            newNum = Long.parseLong(reversedStr);
        } catch (NumberFormatException e) {
            return 0;
        }
        
        // Check for overflow
        if (newNum > Integer.MAX_VALUE) {
            return 0;
        }
        
        // Return the reversed number with the original sign
        return (int) newNum * numSign;
    }
}

File: 7. Reverse Integer/solution.py
class Solution:
    def reverse(self, x: int) -> int:
        # Determine the sign of the number
        numSign = -1 if x < 0 else 1
        
        # Convert the absolute value of the number to a string
        str_x = str(abs(x))
        
        # Reverse the string
        reversed_str_x = str_x[::-1]
        
        # Convert the string back to an integer
        newNum = int(reversed_str_x)
        
        # Check for overflow
        if newNum > 2**31 - 1:
            return 0
        
        # Return the reversed number with the original sign
        return newNum * numSign